army.cpp

```diff
@@ -988,6 +988,22 @@ int Army::NumFront()
 	return (canfront + notfront - canbehind);
 }
 
+int Army::NumBehind() {
+	return NumAlive() - NumFront();
+}
+
+int Army::NumFrontHits() {
+	int totHits = 0;
+
+	for (int i = 0; i < canfront; i++) {
+		totHits += soldiers[i]->maxhits;
+	}
+	for (int i = canbehind; i < notfront; i++) {
```

Comment:
I do not completely understand logic here. Why do you count behind solders towards total front hits?

Reply:
This is taken from Tarmellion. I do not fully follow design of the `Army` class at all with `confront`, `notfront`, etc. But here this won't count behind soldiers :)

This is because:
```
	canfront = x;
	canbehind = count;
	notfront = count;
	notbehind = count;
```

Also the number of units in the front is counted like that: `canfront + notfront - canbehind`.

Also this is a common construction in this class, look to the `GetTargetNum` method...

Taking all this, it looks like there is some logic and the code is valid.